sr_port/mdb_condition_handler.c

```diff
@@ -599,8 +599,24 @@ CONDITION_HANDLER(mdb_condition_handler)
 #	endif
  	err_dev = active_device;
 	active_device = (io_desc *)NULL;
-	dm_action = (frame_pointer->old_frame_pointer->type & SFT_DM)
-		|| (TREF(compile_time) && (frame_pointer->type & SFT_DM));
+	/* Determine if the error occurred on a action from direct mode.
+	 * Go back until we find a counted frame. If we see a SFT_DM frame (which is guaranteed to not be a counted frame)
+	 * before then, we are in direct mode. Else we are not in direct mode.
+	 */
+	start_fp = TREF(compile_time) ? frame_pointer : frame_pointer->old_frame_pointer;
```

Comment:
To answer the commit comment first, the reason that we start at old_frame_pointer here is the if this is the frame of a direct mode command then it is a counted frame which would immediately stop the loop. A counted frame is not necessarily NOT also an indirect frame. When one does an XECUTE command, the generated frame is a counted frame. For that reason, this loop likely has trouble if someone enters the silly indirect command 'XECUTE "<some-command>" '. To verify indirect frames, one really needs to check the stack frame flags for SFF_INDCE which needs to also be worked into this loop.

Reply:
Sorry I don't understand your comment. In my understanding, a direct mode frame (SFT_DM) implies it can never be a counted frame (SFT_COUNT). This is what I am asserting and I tried the XECUTE "" example with this version (debug build which runs the asserts) and no assert failed. It returned back to the direct mode prompt with no output just like it did before. Do you have another test case that would fail? If so, that might help me understand your point.